### PR TITLE
tried a code challenge; please check

### PR DESCRIPTION
**Pull request by spitsfire**
I am blind to TDD with JavaScript, but I wanted acode challenge question, so I tried something new. If it's not worth the sloppy implementation, we can revert it.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -29,3 +29,7 @@ Standards:
         Type: Lesson
         UID: 2a04d88b-f4c9-41eb-be33-6ed68350f222
         Path: /01-js-week-1/js-intro.md
+      - 
+        Type: Lesson
+        UID: c95a8cd5-e90f-42aa-a457-e4162b4f581f
+        Path: /01-js-week-1/jfirst-class-functions.md
```
Comment (CheezItMan) [inline config.yaml]: Typo

```suggestion
        Path: /01-js-week-1/first-class-functions.md
```

File: 01-js-week-1/first-class-functions.md [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+        console.log(`The task name is ${this.name}`);
+        console.log(`The task owner is ${this.owner}`);
+        return true;
+      }
+    }
+    task.describe();
+    
+    expect(testFunction(), "Error message").to.deep.eq(undefined)
+  })
+})
```
Comment (CheezItMan) [inline 01-js-week-1/first-class-functions.md]: This should be a Mocha test that verifies the return value of the function the students entered.  I just can't get anything to pass, including when I paste in your solution from hints.

Maybe have the students `return task` at the end and then test the return value and what's console logged.

I suggest:

```suggestion
/**
 * Blatantly stolen from Jest (I hate mocha/chai with an irrational passion).
 * 
 * This basically lets me mock a function (console.log) so I can see what they printed
 * and how many times they did so.
 * 
 * @param impl - What the real function is supposed to do.
 */
function fn (impl = () => { }) {
  const mockFn = function (...args) {
    mockFn.mock.calls.push(args);
    mockFn.mock.instances.push(this);
    try {
      const value = impl.apply(this, args); // call impl, passing the right this

      mockFn.mock.results.push({ type: 'return', value });
      return value; // return the value
    } catch (value) {
      mockFn.mock.results.push({ type: 'throw', value });
      throw value; // re-throw the error
    }
  }

  mockFn.mock = { calls: [], instances: [], results: [] };
  return mockFn;
}

describe('testFunction', () => {
  let oldConsoleLog = console.log;

  // Mock console.log
  beforeEach(() => {
    console.log = fn();
  });

  // restore console.log
  afterEach(() => {
    console.log = oldConsoleLog;
  });


  it('may look similar to our implementation under hint!', () => {
    const task = testFunction();

    expect(console.log.mock.calls.length).to.equal(2);


    const answer = task.describe();
    expect(answer).to.be.equal(true);

    // The testFunction calls Describe (2 logs) and I call it
    expect(console.log.mock.calls.length).to.equal(4);
    expect(console.log.mock.calls[0][0]).to.equal(`The task name is ${ task.name }`);
    expect(console.log.mock.calls[1][0]).to.equal(`The task owner is ${ task.owner }`);
  });
});
```

**Review by CheezItMan — CHANGES_REQUESTED** [review]
The JavaScript snippet doesn't seem to be working properly for me when I run learn preview, see my suggestion for an alternative question/test.  It would require the testFunction to return `task` as well. 

This also might be heavy in learn questions.

File: 01-js-week-1/first-class-functions.md (hunk 2 above)
Comment (spitsfire) [reply 01-js-week-1/first-class-functions.md]: fair enough! I will fix it up tomorrow

**Comment by dHelmgren** [comment]
@CheezItMan @spitsfire should be in a good way now! We need to give names to the tests so that students can use the test names to figure out what they're doing wrong in the future, debugging this was kind of a pain with one weirdly named test.

**Review by CheezItMan — APPROVED** [review]

**Pull request merged**